lib/normalizer.c

```diff
@@ -1914,12 +1944,9 @@ grn_nfkc_normalize_unify_kana_prolonged_sound_mark(grn_ctx *ctx,
   *n_used_characters = 1;
 
   if (*previous_length == 3 &&
-      /* U+30FC KATAKANA-HIRAGANA PROLONGED SOUND MARK */
-      char_length == 3 &&
-      current[0] == 0xe3 &&
-      current[1] == 0x83 &&
-      current[2] == 0xbc) {
-    previous = current - *previous_length;
+      func(current, char_length)) {
+    const unsigned char *previous = current - *previous_length;
+    *previous_length = char_length;
```

Comment:
How about setting this before `if()` because we should always set it?

Reply:
In order to setting `*previous_length` before this `if`, it is better that we introduce a temporary variable like `saved_previous_length` as below because `*previous_length` is used in the `if()` condition and in brackets of it.

```
  size_t char_length;
  size_t *previous_length = user_data;
  size_t saved_previous_length = *previous_length;

  char_length = (size_t)grn_charlen_(ctx, current, end, GRN_ENC_UTF8);
  *previous_length = char_length;

  *n_used_bytes = char_length;
  *n_used_characters = 1;

  if (saved_previous_length == 3 &&
      func(current, char_length)) {
    const unsigned char *previous = current - saved_previous_length;
    ...
```

How do you feel that? it is better than the current implementation?

Reply:
With https://github.com/groonga/groonga/pull/1526#discussion_r1119658848 , it looks like the following:

```c
size_t previous_length = data->previous_length:
data->previous_length = char_length;

if (previous_length == ...)
```

It looks better than `save_previous_length`.

Reply:
Make sense.

Reply:
Fixed.